pkg/logs/input/kubernetes/launcher.go

```diff
@@ -144,32 +144,57 @@ func (l *Launcher) removeSource(service *service.Service) {
 	}
 }
 
-// kubernetesIntegration represents the name of the integration.
-const kubernetesIntegration = "kubernetes"
+const (
+	// kubernetesIntegration represents the name of the integration.
+	kubernetesIntegration   = "kubernetes"
+	podStandardLabelPrefix  = "tags.datadoghq.com/"
+	tagKeyService           = "service"
+	podStandardLabelService = podStandardLabelPrefix + tagKeyService
+)
 
 // getSource returns a new source for the container in pod.
 func (l *Launcher) getSource(pod *kubelet.Pod, container kubelet.ContainerStatus) (*config.LogSource, error) {
 	var cfg *config.LogsConfig
+	standardServicelabel := l.getServiceLabel(pod, container)
 	if annotation := l.getAnnotation(pod, container); annotation != "" {
 		configs, err := config.ParseJSON([]byte(annotation))
 		if err != nil || len(configs) == 0 {
 			return nil, fmt.Errorf("could not parse kubernetes annotation %v", annotation)
 		}
 		cfg = configs[0]
+		// fmt.Println(cfg.Service)
+		// cfg.Service = "test"
+		if cfg.Service == "" && standardServicelabel != "" {
+			cfg.Service = standardServicelabel
+		}
 	} else {
 		if !l.collectAll {
 			return nil, errCollectAllDisabled
 		}
 		shortImageName, err := l.getShortImageName(container)
```

Comment:
I think you can move this line after line 187, since it is only used in the `else` condition

Reply:
done

Reply:
done